internal/packages/status/status.go

```diff
@@ -0,0 +1,85 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package status
+
+import (
+	"github.com/Masterminds/semver"
+	"github.com/pkg/errors"
+
+	"github.com/elastic/elastic-package/internal/packages"
+	"github.com/elastic/elastic-package/internal/packages/changelog"
+	"github.com/elastic/elastic-package/internal/registry"
+)
+
+// PackageStatus holds version and deployment information about a package
+type PackageStatus struct {
+	Name       string
+	Changelog  []changelog.Revision
+	Local      *packages.PackageManifest
+	Production []packages.PackageManifest
+	Staging    []packages.PackageManifest
+	Snapshot   []packages.PackageManifest
+}
+
+// LocalPackage returns the status of a given package including local development information
+func LocalPackage(packageRootPath string, showAll bool) (*PackageStatus, error) {
+	manifest, err := packages.ReadPackageManifestFromPackageRoot(packageRootPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "reading package manifest failed")
+	}
+	changelog, err := changelog.ReadChangelogFromPackageRoot(packageRootPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "reading package changelog failed")
+	}
+	status, err := Package(manifest.Name, showAll)
+	if err != nil {
+		return nil, err
+	}
+	status.Changelog = changelog
+	status.Local = manifest
+	return status, nil
+}
+
+// Package returns the status of a given package
+func Package(packageName string, showAll bool) (*PackageStatus, error) {
+	snapshotManifests, err := registry.Snapshot.Revisions(packageName, showAll)
```

Comment:
What do you think about transforming `showAll` into something more sophisticated like `ListingOptions`. The options struct can contain all relevant booleans like internal, experimental, showAll. With CLI configuration we can decide which should be enabled by default.

Reply:
I'd be fine with that, but, at least for this use-case the other booleans should always be set, otherwise you're not going to be able to return information for most development level packages. The reason I kept it as `showAll` was due to the fact that the other options don't need to be configurable and only it does. I'm fine refactoring it or keeping it this way until someone leverages this with other options they need to set, at which point we could extract it into an options struct. Just let me know if you'd prefer me do that sooner than later.

Reply:
If it's not a big effort to refactor, I would be go adjusting it here as we don't have many places to adjust. I understand if you prefer to postpone it and I don't want to stop you :)